File: php/src/Builder/SQLBuilder/SqlQueryLogDecorator.php
<?php


namespace Ibrahim13\PdPhp\Builder\SQLBuilder;

use Ibrahim13\PdPhp\Builder\SQLBuilder\SqlQueryDecorator;

class SqlQueryLogDecorator extends SqlQueryDecorator{


	public function __toString(){

		$query = parent::__toString();
		
		file_put_contents('query.log',$query.PHP_EOL,FILE_APPEND );

		return $query;
	}
} 

File: python/builder/sql_query_log_decorator.py
from builder.sql_query_decorator import SqlQueryDecorator
from builder.sql_query import SqlQuery

class SqlQueryLogDecorator(SqlQueryDecorator):


    def __init__(self,sql: str):
       
        super().__init__(sql)
        with open('query.log','a') as file:
            file.write("py#"+str(self.sql_query) +"\n")
        
    


    def __str__(self):
        return str(self.sql_query)

    
